test/tests/neutronics/tally_system/tests

```diff
@@ -0,0 +1,49 @@
+[Tests]
+  [multi_global_estimator]
+    type = Exodiff
+    input = multi_estimator.i
+    exodiff = multi_estimator_out.e
+    requirement = "The system shall correctly normalize local tallies with different estimators using multiple global tallies."
+    required_objects = 'OpenMCCellAverageProblem'
+  []
+  [multiple_tallies]
+    type = Exodiff
+    input = multi.i
+    exodiff = multi_out.e
+    requirement = "The system shall correctly apply and normalize two different local tallies with different scores. "
+                  "The gold file was generated using an input that had a single tally with multiple scores."
+    required_objects = 'OpenMCCellAverageProblem'
+  []
+  [duplicate_scores]
+    type = RunException
+    input = multi.i
+    cli_args = 'Problem/Tallies/Cell_2/tally_score="flux kappa_fission"'
+    expect_err = "You have added 2 tallies which score kappa-fission!"
```

Comment:
What is the motivation for having this error? I could (maybe) see someone interested in having the same score, but with different spatial representation (e.g., cell vs. mesh). It's definitely an edge case and could be fine to keep this as an error, but I'd like to get a better sense on whether there's something fundamental here that I'm missing, or if this is just for convenience?

Reply:
The motivation behind this error is the case where tallies are normalized by the sum of a local tally as opposed to a global tally. If two different tallies share a score they could have overlapping bins, so when the sum of the local tally is computed we might double count tally results. 

This could be alleviated if each tally is normalized by it's own sum as opposed to being normalized by a sum over all of the local tally results for a score. This would lead to complications when trying to accommodate multiple distributed mesh tallies where you need to normalize them by the sum over each "instance" of the tally when performing local normalization; the new tally system is designed such that a single `TallyBase` implies a single `openmc::Tally`. Other complications include how different tallies would store their variables on the mesh mirror if they have the same scores, and which tally's score would be used for normalizing all of the tallies when running in eigenvalue mode (i.e., which `kappa_fission` do you use if both tallies score `kappa_fission`?).

I made the decision to avoid all of these issues by not allowing multiple tallies to share a score, with the compromise that some users wouldn't be able to perform comparisons of different spatial tallies in a single input file.